### PR TITLE
HPCC-20279 Fix roxie core if lockChild gets a blank response

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
@mckellyln please review.  I have not tested the fix.

**Comment by stuartort** [comment]
The fix looks ok if lockResult is allowed to be NULL, otherwise we are addressing the core, but not the real problem (possibly fixing the symptom, not the problem).  This is a good fix for the issue in the next point, but further investigation may be needed

File: roxie/ccd/ccdlistener.cpp [hunk 1]
```diff
@@ -174,7 +174,7 @@ class CascadeManager : public CInterface
         lockQuery.appendf("<control:childlock thisEndpoint='%d' parent='%d'/>", activeIdxes.item(idx), myEndpoint);
         doChildQuery(idx, lockQuery.str(), lockReply);
```
Comment (mckellyln) [inline roxie/ccd/ccdlistener.cpp]: Perhaps we could check for lockReply being empty and log that ?

**Review by mckellyln — APPROVED** [review]
Change is good to prevent null deref.

**Comment by mckellyln** [comment]
@ghalliday looks fine.  @stuartort agree we should know why lockReply is not valid.

**Comment by ghalliday** [comment]
I have opened HPCC-20295 to make sure we also investigate the root cause of the problem

File: roxie/ccd/ccdlistener.cpp (hunk 1 above)
Comment (ghalliday) [reply roxie/ccd/ccdlistener.cpp]: yes, but I would prefer to keep the 6.4 change minimal - improving error reporting can go in master

**Pull request merged**